### PR TITLE
538 - Allow author to add copyright and references for each node

### LINKED ISSUE #538: Allow author to add copyright and references for each node
For the moment only have basic textbox inputs.

### PR DESCRIPTION
**Pull request by narendrasss**
Closes #538 

At the moment this PR only includes textboxes for authors to copy and paste their license/reference information. I expect this will be changed soon as per the [discussion](https://app.asana.com/0/1126491658233864/1132584490660298) on Asana.

Nevertheless this version is ready for testing:

* Open up the node, there should be a "more information" tab in the modal with a copyright and reference text fields
* Write down anything in the copyright text area and submit
* Open the node modal again, the copyright field should be populated

**Note**

I made this PR to merge into #532 to make use of the node modal improvements I did on that PR. Let me know if this should be done into master instead.

### CODE REVIEW

**Comment by liam-armstrong** [comment]
Fields save properly, good to merge

File: classes/class.tapestry-node.php [hunk 1]
```diff
@@ -390,7 +400,9 @@ private function _formNode()
             'fullscreen' => $this->fullscreen,
             'conditions' => $this->conditions,
             'childOrdering' => $this->childOrdering,
-            'fitWindow' => $this->fitWindow,
+            'fitWindow'     => $this->fitWindow,
+            'license'       => $this->license,
+            'references'    => $this->references
```
Comment (wynnset) [inline classes/class.tapestry-node.php:405]: These need PHP linting

**Comment by wynnset** [comment]
I'd like to hold off on merging this until we have the text formatting feature and we can use that for these 2 fields as well.

**Comment by wynnset** [comment]
@narendrasss this task is no longer blocked as the text formatting PR is now merged in. Let's update it with the formatted field. I've asked Simi to provide designs on Asana for adding licenses. Let me know if you need any further clarifications. Thanks!

**Comment by liam-armstrong** [comment]
Text field for the references looks good. But, the license field doesn't seem to be working as intended - the `license` field inside the node definition is still an empty string after setting it and submitting from the Modal. 
A few other comments on the UX:
 - After populating the license field and re-selecting it, the drop down options seem to be filtered with a search. This is a little weird because I expect to see all the license options every time I select it. 
 - There's a plus icon on the `Custom license` option, but pressing it, or selecting it for submission doesn't give me the option to enter a custom license. Not sure what the expected behavior is there.

**Comment by narendrasss** [comment]
@liam-armstrong I think I accidentally addressed these changes in #537 instead, can you try testing that one to see if you're encountering similar problems?

**Comment by liam-armstrong** [comment]
The issues are fixed in #537 , you're right. Maybe you want to move the code into this PR? or merge this PR into 536.

**Comment by narendrasss** [comment]
#537 is actually going into this branch, so I'll merge the sidebar into this. Thanks for testing!

**Comment by wynnset** [comment]
@narendrasss could you please bring the changes over to this PR actually? I think it's better if we merge this into master separately from #537

**Comment by narendrasss** [comment]
I remember doing exactly that but looks like I never pushed! Should be good now.

**Pull request merged**